### ISSUE TITLE
Failing to_file test on pandas master

### ISSUE THREAD

**Issue by jorisvandenbossche** [geopandas/geopandas#711]
```
__________________________ TestDataFrame.test_to_file __________________________
self = <geopandas.tests.test_geodataframe.TestDataFrame instance at 0x7f77f68258c0>
    def test_to_file(self):
        """ Test to_file and from_file """
        tempfilename = os.path.join(self.tempdir, 'boros.shp')
        self.df.to_file(tempfilename)
        # Read layer back in
        df = GeoDataFrame.from_file(tempfilename)
        assert 'geometry' in df
        assert len(df) == 5
        assert np.alltrue(df['BoroName'].values == self.boros)
    
        # Write layer with null geometry out to file
        tempfilename = os.path.join(self.tempdir, 'null_geom.shp')
>       self.df3.to_file(tempfilename)
geopandas/tests/test_geodataframe.py:306: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
geopandas/geodataframe.py:405: in to_file
    to_file(self, filename, driver, schema, **kwargs)
geopandas/io/file.py:106: in to_file
    schema = infer_schema(df)
geopandas/io/file.py:140: in infer_schema
    geom_type = _common_geom_type(df)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
df =   ...
0 ...
1 ...
[2 rows x 2 columns]
    def _common_geom_type(df):
        # Need to check geom_types before we write to file...
        # Some (most?) providers expect a single geometry type:
        # Point, LineString, or Polygon
        geom_types = df.geometry.geom_type.unique()
    
        from os.path import commonprefix
        # use reversed geom types and commonprefix to find the common suffix,
        # then reverse the result to get back to a geom type
>       geom_type = commonprefix([g[::-1] for g in geom_types if g])[::-1]
E       TypeError: 'float' object has no attribute '__getitem__'
geopandas/io/file.py:160: TypeError
```

**Comment by jorisvandenbossche** [comment]
Reason for this: https://github.com/pandas-dev/pandas/issues/20866 (`unique` previously returned None, now NaN, on which the commonprefix fails). Should be easy to fix, but let's first see if we change this back in pandas.

**Comment by ljwolf** [comment]
Does this need to be addressed in geopandas, or are we still waiting on pandas?

**Comment by jorisvandenbossche** [comment]
I think the issue was fixed on the pandas side (and released in 0.23.0), so this should work now.

**Comment by jorisvandenbossche** [comment]
The travis tests on master were actually also green now (at least with the last commit on master), so let's close this. Thanks for pinging here!
